### PR TITLE
Deprecated fields

### PR DESCRIPTION
**Pull request by mo8it**
Fixes https://github.com/idanarye/rust-typed-builder/issues/76

Is there a better test to make sure that a deprecation warning was thrown?

### CODE REVIEW

**Comment by idanarye** [comment]
`lib.rs` has stub function named `_compile_fail_tests`, who's purpose is to have `compile_fail` tests in its documentation comment. You can combine this with disallowing warnings. Something like:

```rust
/// ```compile_fail
/// ...
/// #[disallow(warnings)]
/// something_that_generates_a_warning();
/// ```
```

File: src/lib.rs [hunk 1]
```diff
@@ -182,7 +182,6 @@ fn impl_my_derive(ast: &syn::DeriveInput) -> Result<TokenStream, Error> {
                     .included_fields()
                     .map(|f| struct_info.field_impl(f))
                     .collect::<Result<Vec<_>, _>>()?;
-                let fields = quote!(#(#fields)*).into_iter();
```
[the remark this answers is not in the document]
Comment (idanarye) [reply src/lib.rs:185]: I don't remember why I originally needed this. But since removing it doesn't break the build or the tests it should be okay.

**Review by idanarye — APPROVED** [review]

**Comment by mo8it** [comment]
Thanks! I will move the test there.

**Comment by mo8it** [comment]
@idanarye Done! Thanks again for the test tip, did not know about it!

I did use "outer **block** doc comments" btw: https://doc.rust-lang.org/reference/comments.html#comments

**Comment by mo8it** [comment]
1. You don't have to add `///` for every new line.
2. You save an indentation.

But since this is very subjective, I can just revert if you prefer `///` :)

I did extra mention it as an indirect question if it would be fine for you.

**Comment by idanarye** [comment]
I understand the rationale, but I don't think these benefits are enough to justify diverting from the standard style. Please change it back.

**Comment by mo8it** [comment]
Should be ready to merge now. I did revert The block comments style and resolve conflicts.

**Comment by idanarye** [comment]
Actually, I think this one warrens a changelog entry. I'll write it myself though (no need for a PR just for that)

**Pull request merged**